scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontendState.java

```diff
@@ -1245,12 +1253,33 @@ public UserData getCurrentUserData() throws NotConnectedException {
         return PASchedulerProperties.getPropertiesAsHashMap();
     }
 
+    synchronized ClientJobState getClientJobState(JobId jobId) {
+        if (!jobsMap.containsKey(jobId)) {
+            if (!finishedJobsLRUCache.containsKey(jobId)) {
+                List<InternalJob> internalJobs = dbManager.loadInternalJob(jobId.longValue());
+                if (!internalJobs.isEmpty()) {
+                    InternalJob internalJob = internalJobs.get(0);
+                    ClientJobState clientJobState = new ClientJobState(internalJob);
+                    finishedJobsLRUCache.put(jobId, clientJobState);
+                }
+            }
```

Comment:
If the cache already contains the key, we need to remove it first, then add it to the head of the linkedhashmap